fondant/component.py

```diff
@@ -237,3 +237,57 @@ def _process_dataset(self, manifest: Manifest) -> dd.DataFrame:
         df = self.transform(dataframe=df, **self.user_arguments)
 
         return df
+
+
+class WriteComponent(Component):
+    """Base class for a Fondant write component."""
+
+    @classmethod
+    def _add_and_parse_args(cls, spec: ComponentSpec):
```

Comment:
This is currently the same as the `TransformComponent`, but I guess the output manifest is not required here?

We still have a lot of duplication in these methods for only small differences, so would like to figure out a better way to do this. Maybe we should still create separate schemas?

Reply:
I tackled this in a the PR that follows this by specifying the optional parameters as a specific attribute per component type  

https://github.com/ml6team/fondant/pull/199/files#:~:text=if%20arg.name%20in%20cls.optional_fondant_arguments()%3A 

we could also tackle it [here](https://github.com/ml6team/fondant/blob/56c2dd9f401a2dd17f64ec5dfd572602d3a7e908/fondant/component_spec.py#L211) by having separate secs per component type. This of course would need more reworking and would also required specifying the component type within the component spec yaml (which is not currently the case). Is this what you mean by separate schemas?